### PR TITLE
[14_1_X] Added more ZDC functionality to DQM

### PR DESCRIPTION
**Pull request by cfmcginn**
Code reformatted, emulator updates, and histogram title change

#### PR description:

This is a backport to CMSSW_14_1_X of existing PR here: https://github.com/cms-sw/cmssw/pull/45948

The PR adds to the DQM for ZDC specific histograms, via ZDCQIE10Task

The backport is updated relative to the main PR, mostly fixing build issue, retitling some histograms, and updating python such that emulated ZDC sums fill. It is also rebased to 14_1_X for the backport.

#### PR validation:

PR is tested in CMSSW_14_1_X_2024-10-15-2300, after running a merge-topic to incorporate the PR into a fresh area
scram b ran successfully, validating the build

Following test command produces the expected output:
```
cmsRun DQM/Integration/python/clients/hcal_dqm_sourceclient-live_cfg.py inputFiles=/eos/cms/store/group/dpg_hcal/comm_hcal/PFG/backup_raw/HIHLTPhysics-HIRun2023A-RAW-v1-375754.root runkey=hi_run
```
under the upload directory. Inspected histograms inside the DQM file appear reasonable for ZDC sums.

In addition, the following set of commands are run
scram b runtests
scram build code-checks
scram build code-format

code-checks revealed no issues, code-format suggestions were adopted. runtests had multiple failures, detailed below - however, every failure existed already with runtests in a fresh area of CMSSW_14_1_X_2024-10-15-2300 with the same packages checked out (before merge-topic), so it is unclear it is the result of the PR

```
>> Local Products Rules ..... started
>> Local Products Rules ..... done
Creating test log file logs/el9_amd64_gcc12/testing.log
Fail   26s ... DQM/Integration/TestDQMOnlineClient-beam_dqm_sourceclient
Pass   38s ... DQM/Integration/TestDQMOnlineClient-beamhlt_dqm_sourceclient
Fail    6s ... DQM/Integration/TestDQMOnlineClient-beampixel_dqm_sourceclient
Fail   49s ... DQM/Integration/TestDQMOnlineClient-csc_dqm_sourceclient
Fail    7s ... DQM/Integration/TestDQMOnlineClient-ctpps_dqm_sourceclient
Fail    6s ... DQM/Integration/TestDQMOnlineClient-dt4ml_dqm_sourceclient
Fail    6s ... DQM/Integration/TestDQMOnlineClient-dt_dqm_sourceclient
Fail    6s ... DQM/Integration/TestDQMOnlineClient-ecal_dqm_sourceclient
Pass   17s ... DQM/Integration/TestDQMOnlineClient-ecalgpu_dqm_sourceclient
Fail    7s ... DQM/Integration/TestDQMOnlineClient-es_dqm_sourceclient
Fail    6s ... DQM/Integration/TestDQMOnlineClient-fed_dqm_sourceclient
Fail    7s ... DQM/Integration/TestDQMOnlineClient-gem_dqm_sourceclient
Fail    6s ... DQM/Integration/TestDQMOnlineClient-hcal_dqm_sourceclient
Pass   10s ... DQM/Integration/TestDQMOnlineClient-hcalgpu_dqm_sourceclient
Fail    6s ... DQM/Integration/TestDQMOnlineClient-hcalreco_dqm_sourceclient
Fail    6s ... DQM/Integration/TestDQMOnlineClient-hlt_dqm_clientPB
Fail    6s ... DQM/Integration/TestDQMOnlineClient-hlt_dqm_sourceclient
Fail    5s ... DQM/Integration/TestDQMOnlineClient-info_dqm_sourceclient
Fail    6s ... DQM/Integration/TestDQMOnlineClient-l1tstage2_dqm_sourceclient
Fail    6s ... DQM/Integration/TestDQMOnlineClient-l1tstage2emulator_dqm_sourceclient
Fail    6s ... DQM/Integration/TestDQMOnlineClient-mutracking_dqm_sourceclient
Pass    9s ... DQM/Integration/TestDQMOnlineClient-onlinebeammonitor_dqm_sourceclient
Pass    7s ... DQM/Integration/TestDQMOnlineClient-pfgpu_dqm_sourceclient
Fail    6s ... DQM/Integration/TestDQMOnlineClient-pixel_dqm_sourceclient
Pass   10s ... DQM/Integration/TestDQMOnlineClient-pixelgpu_dqm_sourceclient
Fail    5s ... DQM/Integration/TestDQMOnlineClient-pixellumi_dqm_sourceclient
Fail    6s ... DQM/Integration/TestDQMOnlineClient-rpc_dqm_sourceclient
Fail    5s ... DQM/Integration/TestDQMOnlineClient-scal_dqm_sourceclient
Pass   33s ... DQM/Integration/TestDQMOnlineClient-sistrip_approx_dqm_sourceclient
Fail    6s ... DQM/Integration/TestDQMOnlineClient-sistrip_dqm_sourceclient
Fail    6s ... DQM/Integration/TestDQMOnlineClient-visualization
Fail    6s ... DQM/Integration/TestDQMOnlineClient-visualization_secondInstance
Pass  172s ... DQM/Integration/TestDQMOnlineClients-compilation_commissioning_run
Pass  142s ... DQM/Integration/TestDQMOnlineClients-compilation_cosmic_run
Pass  145s ... DQM/Integration/TestDQMOnlineClients-compilation_cosmic_run_stage1
Pass  155s ... DQM/Integration/TestDQMOnlineClients-compilation_hi_run
Pass  152s ... DQM/Integration/TestDQMOnlineClients-compilation_hpu_run
Pass  149s ... DQM/Integration/TestDQMOnlineClients-compilation_pp_run
Pass  131s ... DQM/Integration/TestDQMOnlineClients-compilation_pp_run_stage1
>> Test sequence completed for CMSSW CMSSW_14_1_X_2024-10-15-2300
```

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

As mentioned above, PR is a backport (with modification) of https://github.com/cms-sw/cmssw/pull/45948, intended for CMSSW_14_1_X

One note - close inspection of the changes to the python file would be welcome - the changes were made as part of testing the full package but unclear to me all should be incorporated into the branch as is. Guidance appreciated.

@hjbossi @mandrenguyen

### CODE REVIEW

**Comment by mandrenguyen** [comment]
please test

**Comment by hjbossi** [comment]
Just to add another tag for relevant reviewers: @denizsun, @salimcerci, @JoshPBR2, @abdoulline. This is a modification of of the original PR, which we intend to update with the changes once @JoshPBR2 is available to do so. The current version of the release is 14_1_3, built yesterday. If possible, we would love for this to be included the next release, i.e. 14_1_4. 

As @cfmcginn mentions, it would be good to understand what is most appropriate to do for the source client file. The reason why we have questions is that I am not entirely sure what is used to create the plots online vs. local tests. Please let us know what you think!

File: DQM/Integration/python/clients/hcal_dqm_sourceclient-live_cfg.py [hunk 1]
```diff
@@ -94,27 +97,34 @@
...
+
 process.hcalDigis.InputLabel = rawTag
 process.emulTPDigisNoTDCCut = process.emulTPDigis.clone(
      parameters = cms.untracked.PSet(
 	ADCThresholdHF = cms.uint32(255),
 	TDCMaskHF = cms.uint64(0xFFFFFFFFFFFFFFFF)
      )
 )
-process.HcalTPGCoderULUT.LUTGenerationMode = False
+process.HcalTPGCoderULUT.LUTGenerationMode = True
```
Comment (mmusich) [inline DQM/Integration/python/clients/hcal_dqm_sourceclient-live_cfg.py:127]: what's the purpose of this change?

File: DQM/Integration/python/clients/hcal_dqm_sourceclient-live_cfg.py [hunk 2]
```diff
@@ -243,6 +254,10 @@
 )
 process.options.wantSummary = True
 
+process.maxEvents = cms.untracked.PSet(
+    input = cms.untracked.int32(-1)
+    )
```
Comment (mmusich) [inline DQM/Integration/python/clients/hcal_dqm_sourceclient-live_cfg.py:259]:
```suggestion
```

in live mode the number of events processed is regulated elsewhere:

https://github.com/cms-sw/cmssw/blob/f97584b622ab5e783f4fe2a50611c1e269b65629/DQM/Integration/python/config/inputsource_cfi.py#L111-L125

**Review by mmusich — COMMENTED** [review]
This PR is different from the master one: https://github.com/cms-sw/cmssw/pull/45948, thus it's not a backport. You need to close that one and open another one that contains the same changes as this one in the master (CMSSW_14_2_X) cycle.

**Comment by hjbossi** [comment]
> This PR is different from the master one: #45948, thus it's not a backport. You need to close that one and open another one that contains the same changes as this one in the master (CMSSW_14_2_X) cycle.

Hi @mmusich, thanks for your comment! The plan is to update the PR to master  (CMSSW_14_2_X) cycle to have the same changes as the one here, i.e. this would be a true backport. Unfortunately, we've been unable to get ahold of Josh (I believe he might be traveling to CERN), and he is needed in order to make this change.  We wanted to give ample time for review of these changes, so we decided to open the PR to 14_1_X as soon as possible. Let me know what you think of this strategy.

**Comment by mmusich** [comment]
>  Let me know what you think of this strategy.

well, abandoning the unattended PR and opening a new one from scratch will get you merged faster. This one cannot be integrated until the master one is merged.

File: DQM/Integration/python/clients/hcal_dqm_sourceclient-live_cfg.py (hunk 1 above)
Comment (hjbossi) [reply DQM/Integration/python/clients/hcal_dqm_sourceclient-live_cfg.py:127]: This is actually what we would like feedback on from @denizsun and @salimcerci, this is necessary for the code to run with the candidate GT (which does not yet include an updated L1TriggerObject) , but probably not necessary for actual data-taking. In this case, should we include this or leave it out?

File: DQM/Integration/python/clients/hcal_dqm_sourceclient-live_cfg.py [hunk 3]
```diff
@@ -32,10 +32,11 @@
 #-------------------------------------
 from DQM.Integration.config.online_customizations_cfi import *
 if useOfflineGT:
-	process.load("Configuration.StandardSequences.FrontierConditions_GlobalTag_cff")
-	process.GlobalTag.globaltag = autoCond['run3_data_prompt'] 
+        process.load("Configuration.StandardSequences.FrontierConditions_GlobalTag_cff")
+        process.GlobalTag.globaltag = '141X_dataRun3_Prompt_Candidate_2024_10_08_09_42_50'
```
[the remark this answers is not in the document]
Comment (hjbossi) [reply DQM/Integration/python/clients/hcal_dqm_sourceclient-live_cfg.py:36]: Sounds good, the above is a candidate GT, but the default version will change once the new GT (containing the changes from the candidate) is in place.

**Comment by hjbossi** [comment]
> > Let me know what you think of this strategy.
> 
> well, abandoning the unattended PR and opening a new one from scratch will get you merged faster. This one cannot be integrated until the master one is merged.

Okay I see, in that case we will see if Josh gets back to us today, and if not, we will close the master PR and open a new one.

**Comment by antoniovagnerini** [comment]
> > > Let me know what you think of this strategy.
> > 
> > 
> > well, abandoning the unattended PR and opening a new one from scratch will get you merged faster. This one cannot be integrated until the master one is merged.
> 
> Okay I see, in that case we will see if Josh gets back to us today, and if not, we will close the master PR and open a new one.

@hjbossi any updates on the comments by Marco? At this stage, I would also recommend closing the master https://github.com/cms-sw/cmssw/pull/45948 and making a new one that matches the backport.

**Comment by cfmcginn** [comment]
> > > > Let me know what you think of this strategy.
> > > 
> > > 
> > > well, abandoning the unattended PR and opening a new one from scratch will get you merged faster. This one cannot be integrated until the master one is merged.
> > 
> > 
> > Okay I see, in that case we will see if Josh gets back to us today, and if not, we will close the master PR and open a new one.
> 
> @hjbossi any updates on the comments by Marco? At this stage, I would also recommend closing the master #45948 and making a new one that matches the backport.

I intend to open the new PR to main today; I don't know that I have power to close the old, but that should follow

**Comment by hjbossi** [comment]
Looking at the master PR https://github.com/cms-sw/cmssw/pull/45948, I don't think I have permissions to close it, which means that must be done by @JoshPBR2. Although maybe someone with less restricted permissions can do so?

**Comment by cfmcginn** [comment]
PR to main can be found here:

https://github.com/cms-sw/cmssw/pull/46478

File: DQM/HcalTasks/interface/ZDCQIE10Task.h [hunk 4]
```diff
@@ -9,24 +9,87 @@
...
 public:
-  ZDCQIE10Task(edm::ParameterSet const&);
+  ZDCQIE10Task(edm::ParameterSet const &);
   ~ZDCQIE10Task() override {}
 
-  void bookHistograms(DQMStore::IBooker&, edm::Run const&, edm::EventSetup const&) override;
+  void bookHistograms(DQMStore::IBooker &, edm::Run const &, edm::EventSetup const &) override;
+  //    std::shared_ptr<hcaldqm::Cache> globalBeginLuminosityBlock(edm::LuminosityBlock const &,
+  //							       edm::EventSetup const &) const override;
+  //void globalEndLuminosityBlock(edm::LuminosityBlock const &, edm::EventSetup const &) override;
```
[the remark this answers is not in the document]
Comment (cfmcginn) [reply DQM/HcalTasks/interface/ZDCQIE10Task.h]: No, removed in most recent commit

File: DQM/HcalTasks/interface/ZDCQIE10Task.h [hunk 5]
```diff
@@ -9,24 +9,87 @@
...
 
   //	tags
   edm::InputTag _tagQIE10;
+  edm::InputTag sumTag;
   edm::EDGetTokenT<QIE10DigiCollection> _tokQIE10;
+  edm::ESGetToken<HcalDbService, HcalDbRecord> hcalDbServiceToken_;
+  edm::EDGetToken sumToken_;
+  edm::ESGetToken<HcalTopology, HcalRecNumberingRecord> htopoToken_;
+  edm::ESGetToken<HcalLongRecoParams, HcalLongRecoParamsRcd> paramsToken_;
+  //edm::EDGetTokenT<l1t::EtSumBxCollection> sumZDCToken_;
```
[the remark this answers is not in the document]
Comment (cfmcginn) [reply DQM/HcalTasks/interface/ZDCQIE10Task.h]: No, removed in most recent commit

File: DQM/HcalTasks/plugins/ZDCQIE10Task.cc [hunk 6]
```diff
@@ -1,116 +1,529 @@
...
+ZDCQIE10Task::ZDCQIE10Task(edm::ParameterSet const& ps)
 
-ZDCQIE10Task::ZDCQIE10Task(edm::ParameterSet const& ps) {
-  //	tags
+    : DQTask(ps), hcalDbServiceToken_(esConsumes<HcalDbService, HcalDbRecord, edm::Transition::BeginRun>()) {
+  //tags
   _tagQIE10 = ps.getUntrackedParameter<edm::InputTag>("tagQIE10", edm::InputTag("hcalDigis", "ZDC"));
   _tokQIE10 = consumes<QIE10DigiCollection>(_tagQIE10);
+
+  //const edm::EDGetTokenT<l1t::EtSumBxCollection> sumZDCToken_;
```
[the remark this answers is not in the document]
Comment (cfmcginn) [reply DQM/HcalTasks/plugins/ZDCQIE10Task.cc]: Adopted

File: DQM/HcalTasks/plugins/ZDCQIE10Task.cc [hunk 7]
```diff
@@ -1,116 +1,529 @@
...
+
+  sumTag = ps.getUntrackedParameter<edm::InputTag>("etSumTag", edm::InputTag("etSumZdcProducer", ""));
+  //sumTag = ps.getParameter<edm::InputTag>("etSumTag");
```
[the remark this answers is not in the document]
Comment (cfmcginn) [reply DQM/HcalTasks/plugins/ZDCQIE10Task.cc]: Adopted

File: DQM/HcalTasks/plugins/ZDCQIE10Task.cc [hunk 8]
```diff
@@ -1,116 +1,529 @@
...
+  sumToken_ = consumes<l1t::EtSumBxCollection>(sumTag);
+  //sumZDCToken_ = consumes<l1t::EtSumBxCollection>(ps.getUntrackedParameter<edm::InputTag>("sumZDCToken"));
```
[the remark this answers is not in the document]
Comment (cfmcginn) [reply DQM/HcalTasks/plugins/ZDCQIE10Task.cc]: Adopted

File: DQM/Integration/python/clients/hcal_dqm_sourceclient-live_cfg.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (cfmcginn) [reply DQM/Integration/python/clients/hcal_dqm_sourceclient-live_cfg.py]: This is reverted in most recent commit

File: DQM/Integration/python/clients/hcal_dqm_sourceclient-live_cfg.py [hunk 9]
```diff
@@ -57,10 +58,12 @@
 process.dqmSaverPB.runNumber = options.runNumber
 process = customise(process)
 process.DQMStore.verbose = 0
+
+'''
```
[the remark this answers is not in the document]
Comment (cfmcginn) [reply DQM/Integration/python/clients/hcal_dqm_sourceclient-live_cfg.py]: Done, no longer commented in most recent commit

File: DQM/Integration/python/clients/hcal_dqm_sourceclient-live_cfg.py (hunk 2 above)
Comment (cfmcginn) [reply DQM/Integration/python/clients/hcal_dqm_sourceclient-live_cfg.py]: Removed in most recent commit

**Comment by antoniovagnerini** [comment]
backport of https://github.com/cms-sw/cmssw/pull/46478

**Comment by antoniovagnerini** [comment]
please test

**Comment by cfmcginn** [comment]
Not all of the requested changes in https://github.com/cms-sw/cmssw/pull/46478 have been incorporated here (because of the history of how this was setup) - I will incorporate them now (just the last commit including some header removals and reversion of a bool to false, can provide more details as needed)

**Comment by antoniovagnerini** [comment]
please abort

**Comment by cfmcginn** [comment]
Backport is now up-to-date w/ main merge, ready for testing. Note commits are not squashed, can squash at request

**Comment by antoniovagnerini** [comment]
please test

**Comment by antoniovagnerini** [comment]
+1

**Comment by mandrenguyen** [comment]
+1
